constrained_by_test.go

```diff
@@ -0,0 +1,40 @@
+package worksheets
+
+import (
+	"strings"
+
+	"github.com/stretchr/testify/require"
+)
+
+func (s *Zuite) TestWorksheet_constrainedBy() {
+	defs, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		1:name text constrained_by { return name == "Alex" || name == "Wilson" }
+	}`))
+	require.NoError(s.T(), err)
+
+	ws := defs.MustNewWorksheet("simple")
+
+	require.False(s.T(), ws.MustIsSet("name"))
+	err = ws.Set("name", NewText("Alice"))
+	require.Equal(s.T(), "\"Alice\" not a valid value for constrained field name", err.Error())
```

Comment:
nit: nicer to use a non-interpolated string for that

    `"Alice" is not a valid value for constrained field name`